network/common/src/test/java/org/apache/spark/network/buffer/LargeByteBufferHelperSuite.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network.buffer;
+
+import java.io.*;
+import java.nio.channels.FileChannel;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class LargeByteBufferHelperSuite {
+
+  @Test
+  public void testMapFile() throws IOException {
+    File testFile = File.createTempFile("large-byte-buffer-test", ".bin");
+    try {
+      testFile.deleteOnExit();
+      OutputStream out = new FileOutputStream(testFile);
+      byte[] buffer = new byte[1 << 16];
+      long len = ((long)buffer.length) + Integer.MAX_VALUE + 1;
```

Comment:
I can certainly add some test that is like the memory-based one which will be smaller.  But I put this in specifically because I want *at least one* test that stresses behavior over 2GB.  I figure the best way to do with large files that get memory mapped.

I don't love this either, but don't you think we should have at least some test with over 2GB, since that's the whole point of the patch?  Any other ideas of how we can do it?